lib/src/providers/TranslationProvider/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { PropsWithChildren, useMemo } from 'react';
+import { createIntl, createIntlCache, RawIntlProvider, type ResolvedIntlConfig } from 'react-intl';
+import resourceLoader, { type ResourceLoader } from '../../utils/suspense/resourceLoader';
+import { VERNA_SUPPORTED_LOCALES } from '../../settings';
+import { TranslationType } from '../../types/translations';
+
+interface TranslationProviderProps {
+  defaultLocale?: string;
+  intl?: ResolvedIntlConfig;
+  locale?: string;
+  schemaTranslations?: TranslationType;
```

Comment:
```suggestion
  defaultLocale?: string;
  intl?: ResolvedIntlConfig;
  locale?: string;
  schemaTranslations?: TranslationType;
```
is there a way not to have all parameters optional?

Reply:
I would say those would be mandatory and then put the whole provider optional in the other parts of the app (we will see further friday :) )

Reply:
I tried to find a better type but I did not get it.

I was looking for something like that `{intl: ResolvedIntlConfig} | {defaultLocale?:string, locale:string}` but it did not work as expected.